### PR TITLE
feat(checkboxes): Added checkbox support for Markdown.

### PR DESCRIPTION
**Pull request by michaelkleinhenz**
Adds checkbox support for Markdown.

### CODE REVIEW

File: src/app/github-link-area/examples/github-link-area-example.component.ts [hunk 1]
```diff
@@ -20,16 +20,30 @@ import {
 })
 export class GitHubLinkAreaExampleComponent implements OnInit {
 
-  content: string = `There is some text in here
-    <a href="https://github.com/patternfly/patternfly-ng/issues/127">https://github.com/patternfly/patternfly-ng/issues/127</a>. 
+  content: string = `There is some text in here.
+    And some checkboxes:<ul>
+    <li><input type="checkbox" data-checkbox-index="0"></input>An Item.</li>
+    <li><input type="checkbox" checked="" data-checkbox-index="1"></input>Checked Item.</li>
```
Comment (nimishamukherjee) [inline src/app/github-link-area/examples/github-link-area-example.component.ts]: @michaelkleinhenz what should checked="false" do ?
<input type="checkbox" checked="false" data-checkbox-index="1"></input>Checked Item.</li>

Comment (michaelkleinhenz) [reply src/app/github-link-area/examples/github-link-area-example.component.ts]: Leftover. I'll remove that. :-)

File: src/app/github-link-area/safe.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ public transform(value: any, type: string):
+  SafeHtml | SafeStyle | SafeScript | SafeUrl | SafeResourceUrl {
+    switch (type) {
+      case 'html': return this.sanitizer.bypassSecurityTrustHtml(value);
+      case 'style': return this.sanitizer.bypassSecurityTrustStyle(value);
+      case 'script': return this.sanitizer.bypassSecurityTrustScript(value);
+      case 'url': return this.sanitizer.bypassSecurityTrustUrl(value);
+      case 'resourceUrl': return this.sanitizer.bypassSecurityTrustResourceUrl(value);
+      default: throw new Error(`Invalid safe type specified: ${type}`);
+    }
```
Comment (dlabrecq) [inline src/app/github-link-area/safe.pipe.ts]: Could be wrong, but this object appears to disable/bypass Angular's built-in sanitation? If so, bypassing JavaScript sanitation isn't a good idea.

I'm concerned we're converting user input into a trusted value. I worry that we're introducing a security vulnerability by trusting values that could be malicious?

See: https://angular.io/guide/security#bypass-security-apis
`"But be careful. If you trust a value that might be malicious, you are introducing a security vulnerability into your application. If in doubt, find a professional security reviewer."`

**Comment by joshuawilson** [comment]
@michaelkleinhenz does your new PR address the issue that @dlabrecq raised?

**Comment by michaelkleinhenz** [comment]
@joshuawilson @dlabrecq yes and no. The wrapping in SafeValue is now happening outside of ngx-widgets in the enclosing component (namely the Planner detail view). ngx-widgets now only just supports incoming SafeValue content. All sanitizer handling has been removed from ngx-widgets.

Although the problem still persists, because we still need to use SafeValue in Planner to wrap the rendered markdown. The Angular sanitizer can not be configured on a fine granular (element) basis (like the golang sanitizer can). So we either need to disable HTML sanitizing entirely for that value or we can not support any feature that needs input elements in the HTML. Note that the HTML sanitizing can be individually turned off, JavaScript and URLs are still stripped from the HTML. 

Although this is still not ideal, the impact there is low. We're only turning off HTML sanitizing, other active components are still stripped. The HTML comes from a trusted source and is not directly user-editable but is always generated from Markdown. Also, the generated HTML is sanitized on the server side as well in a much more restricted way (*only* allowing <input> elements in addition to the standard set of allowed elements).

File: src/app/github-link-area/safe.pipe.ts (hunk 2 above)
Comment (michaelkleinhenz) [reply src/app/github-link-area/safe.pipe.ts]: See comment below.

**Comment by dlabrecq** [comment]
I confirmed the github-link-area component is still bypassing HTML, while the markdown example is bypassing HTML, styles, JavaScript, URLs, etc.

Is the intention to simply move the bypass code (in the example) to Planner?

```
./app/github-link-area/github-link-area.component.ts:      return this.sanitizer.bypassSecurityTrustHtml(input);
./app/markdown/examples/markdown-example.component.ts:      case 'html': return this.sanitizer.bypassSecurityTrustHtml(value);
./app/markdown/examples/markdown-example.component.ts:      case 'style': return this.sanitizer.bypassSecurityTrustStyle(value);
./app/markdown/examples/markdown-example.component.ts:      case 'script': return this.sanitizer.bypassSecurityTrustScript(value);
./app/markdown/examples/markdown-example.component.ts:      case 'url': return this.sanitizer.bypassSecurityTrustUrl(value);
./app/markdown/examples/markdown-example.component.ts:      case 'resourceUrl': return this.sanitizer.bypassSecurityTrustResourceUrl(value);
./app/markdown/examples/markdown-example.component.ts:      callBack(rawText, this.sanitizer.bypassSecurityTrustHtml(text));
./app/markdown/markdown.component.spec.ts:          comp.inpRenderedText = domSanitizer.bypassSecurityTrustHtml(originalHTML);
```

**Comment by dlabrecq** [comment]
Yes. As long as the JavaScript and URLs are still stripped from the HTML, I'm fine with the markup being sanitized server-side.

**Pull request merged**